design-patterns/structural-patterns/src/decorator/Application.java

```diff
@@ -0,0 +1,16 @@
+package decorator;
+
+public class Application {
+
+  public static void main( String[] args ) {
+	  Sandwich mySandwich = new VegetarianSandwich();
+	  SauceSandwich mySandwichWithSauce = new SauceSandwich(mySandwich);
+	  
+	  SauceSandwich mySandwichWithDoubleSauce = new SauceSandwich(new SauceSandwich(mySandwich));
+	  mySandwichWithDoubleSauce.makeWith("mustard");
```

Comment:
Try to add add 1 more sauce to the sandwich. I believe this wont work as expected

Reply:
Fixed.